application_handlers.go

```diff
@@ -233,3 +233,61 @@ func SourceListApplications(c echo.Context) error {
 
 	return c.JSON(http.StatusOK, util.CollectionResponse(out, c.Request(), int(count), limit, offset))
 }
+
+// ApplicationPause pauses a given application by setting its "paused_at" column to "now()".
+func ApplicationPause(c echo.Context) error {
+	applicationId, err := strconv.ParseInt(c.Param("id"), 10, 64)
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, util.ErrorDoc(err.Error(), "400"))
```

Comment:
Here you can return util.NewErrBadRequest(err). Same below and in source_handler.go. HandleErrors() in middleware recognise err type and prepare JSON with correct status and message.

Reply:
Thank you, nice catch!

Reply:
not fixed yet 😁